pkg/util/clusterdata/clusterdata.go

```diff
@@ -32,6 +32,8 @@ type ParallelEnricher struct {
 	enrichers map[string]ClusterEnricher
 	emitter   metrics.Emitter
 	dialer    proxy.Dialer
+	//only used for testing because the metrics emission is async
+	metricsWG *sync.WaitGroup
```

Comment:
though, it is probably better than adding sleep statements in the test?

Reply:
so here since it is async I am not sure we have a better way in any case, since adding sleep statement in tests isn't exactly good practice. another way would have been to only send the error to the channel after the metrics has been emitted but that would make some weird issues happening if the metrics channel is full and we need to wait for it 

so yeah, it isn't optimal... but it's the least bad of the solutions imo